cosmos/airflow/graph.py

```diff
@@ -409,6 +409,27 @@ def _get_dbt_dag_task_group_identifier(dag: DAG, task_group: TaskGroup | None) -
     return dag_task_group_identifier
 
 
+def identify_detached_nodes(
+    nodes: dict[str, DbtNode],
+    test_behavior: TestBehavior,
+    detached_nodes: dict[str, DbtNode],
+    detached_from_parent: dict[str, list[DbtNode]],
+) -> None:
+    """
+    Given the nodes that represent a dbt project and the test_behavior, identify the detached test nodes
+    (test nodes that have multiple dependencies and should run independently).
+
+    Change in-place the dictionaries detached_nodes (detached node ID : node) and detached_from_parent (parent node ID that
+    is upstream to this test and the test node).
+    """
+    if test_behavior in (TestBehavior.BUILD, TestBehavior.AFTER_EACH):
+        for node_id, node in nodes.items():
+            if is_detached_test(node):
+                detached_nodes[node_id] = node
+                for parent_id in node.depends_on:
+                    detached_from_parent[parent_id].append(node)
```

Comment:
```suggestion
    if test_behavior not in (TestBehavior.BUILD, TestBehavior.AFTER_EACH):
        return
    for node_id, node in nodes.items():
        if not is_detached_test(node):
            continue
        detached_nodes[node_id] = node
        for parent_id in node.depends_on:
            detached_from_parent[parent_id].append(node)
```

suggestion to reduce a couple of levels of complexity if you like it based on https://medium.com/swlh/return-early-pattern-3d18a41bba8 but I know everyone may not feel the same way :)

Reply:
Thanks, @pankajkoti ! Since it's within our accepted complexity levels, I suggest we keep, but it is certainly worth to keep this in mind if we want to reduce the complexity level further.